src/player.cpp

```diff
@@ -3626,6 +3628,45 @@ void player::apply_damage(Creature *source, body_part hurt, int dam)
     if( dam > get_painkiller() ) {
         on_hurt( source );
     }
+
+    // remove healing effects if damaged
+    int remove_med = roll_remainder( dam / 5.0f );
+    int bandaged_intensity = 0;
+    int disinfected_intensity = 0;
+
+    if( remove_med > 0 && has_effect( effect_bandaged, hurt ) ) {
+        auto matching_map_bandaged = effects->find( effect_bandaged );
```

Comment:
Why are you accessing the `effects` map directly instead of using the function in `Creature`, like `get_effect`?

Reply:
Because of my low skill. It shoul look beter now.